cli/pcluster/utils.py

```diff
@@ -131,21 +131,35 @@ def _get_json_from_s3(region, file_name):
     return json.loads(file_contents)
 
 
-def get_supported_batch_instances(region):
+def get_supported_features(region, feature):
     """
-    Get a json object containing the instances supported by batch.
+    Get a json object containing the attributes supported by a feature, for example.
+
+    {
+        "Features": {
+            "efa": {
+                "instances": ["c5n.18xlarge", "p3dn.24xlarge", "i3en.24xlarge"],
+                "baseos": ["alinux", "centos7"],
+                "schedulers": ["sge", "slurm", "torque"]
+            },
+            "awsbatch": {
+                "instances": ["r3.8xlarge", ..., "m5.4xlarge"]
+            }
+        }
+    }
 
     :param region: AWS Region
-    :param instance_type: the instance type to search for.
-    :return: json object containing the instances supported by batch
+    :param feature: the feature to search for, i.e. "efa" "awsbatch"
+    :return: json object containing all the attributes supported by feature
     or an empty object if unable to parse/get the instance list file
     """
     try:
-        instances = _get_json_from_s3(region, "instances/batch_instances.json")
+        features = _get_json_from_s3(region, "features/feature_whitelist.json")
+        supported_features = features.get("Features").get(feature)
     except (ValueError, ClientError):
-        instances = ""
+        supported_features = {}
```

Comment:
I would actually fail with unable to retrieve... rather than returning empty object otherwise it seems the validation failed for some other unmet requirement.
Also shall we catch the broader Exception so that we also consider KeyError?

Reply:
Ok, I've changed the exception logic to give customers a workaround should something fail. This is coupled with the above change to have all calls to `get_supported_features()` in config_sanity.py so it can be disabled if anything breaks.